### PR TITLE
Vignette edits

### PR DESCRIPTION
**Pull request by t-kalinowski**
This patch also updates all the optimizers to use `learning_rate` instead of `lr`.

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -398,3 +398,8 @@ is_keras_tensor <- function(x) {
     k_is_tensor(x)
   }
 }
+
+
+# conflict with utils::zip, maybe another name?
+zip <- function(..., simplify = TRUE)
```
Comment (skeydan) [inline R/utils.R:404]: does `py_zip` make sense ...? or even .... move this to `reticulate`?

File: vignettes/guides/customizing_what_happens_in_fit.Rmd [hunk 2]
```diff
@@ -32,6 +32,13 @@ editor_options:
 library(keras)
 ```
 
+## Summary
+
+In this guide you learn how to customize what happens when you call `fit(model)`.
```
Comment (skeydan) [inline vignettes/guides/customizing_what_happens_in_fit.Rmd:37]:
```suggestion
In this guide you learn how to customize what happens when you call `fit(model)`. This functionality presupposes that you define your own custom model.
```

File: vignettes/guides/customizing_what_happens_in_fit.Rmd [hunk 3]
```diff
@@ -32,6 +32,13 @@ editor_options:
...
+You see how to subclass `keras$Model` and create your own custom keras Model,
```
Comment (skeydan) [inline vignettes/guides/customizing_what_happens_in_fit.Rmd:38]:
```suggestion
To that end, you see how to subclass `keras$Model`
```

File: vignettes/guides/customizing_what_happens_in_fit.Rmd [hunk 4]
```diff
@@ -32,6 +32,13 @@ editor_options:
...
+with it's own custom `train_step()` method.
```
Comment (skeydan) [inline vignettes/guides/customizing_what_happens_in_fit.Rmd:39]:
```suggestion
, defining a custom `train_step()` method that does what you need.
```

**Review by skeydan — APPROVED** [review]
great, thanks for asking! Left a few suggestions.

File: R/utils.R (hunk 1 above)
Comment (t-kalinowski) [reply R/utils.R]: I would expect `py_zip` to be a wrapper around `reticulate::import_builtins()$zip`.

**Pull request merged**